### ISSUE TITLE
Use ISO 8601 and or UTC/Offset as default date format

### ISSUE THREAD

**Issue by leonard84** [ktoso/maven-git-commit-id-plugin#288]
The current default date format `dd.MM.yyyy '@' HH:mm:ss z` uses locale dependent zone ids, which are difficult to parse correctly. So please either use UTC or the offset style.

* German Locale: `MEZ` and `MESZ`
* English Localse `CET` and `CEST`

So to avoid having to use try-catch cascades to parse this date in java we should just use the offset variant, preferably the ISO 8601 since it is well defined and already has the appropriate parsers in most environments:

* `dd.MM.yyyy '@' HH:mm:ss Z` ->`03.12.2011 @ 10:15:30:00 +0000` `03.12.2011 @ 12:15:30:00 +0200`
* `dd.MM.yyyy '@' HH:mm:ss X` ->`03.12.2011 @ 10:15:30:00 Z`  `03.12.2011 @ 12:15:30:00 +0200`
* ISO 8601 compliant format `yyyy-MM-dd'T'HH:mmX` -> `2011-12-03T10:15:30Z`  `2011-12-03T12:15:30+0200`

**Comment by TheSnoozer** [comment]
Let me try to elaborate here.....kinda agree that we messed up the default date format. However the plugin allows to overwrite the timestap with the configuration option ```<dateFormat>dd.MM.yyyy '@' HH:mm:ss z</dateFormat>``` (see full example below). You can easily adjust it to your needs.

I'm not sure if someone who is already using the plugin has some parser that already parses the current timestamp format and thus changing the plugin's default would break this feature....
Regardless if I would change it I would prefer Maven's default (https://maven.apache.org/guides/introduction/introduction-to-the-pom.html#Available_Variables) is:
```<maven.build.timestamp.format>yyyy-MM-dd'T'HH:mm:ss'Z'</maven.build.timestamp.format>```

I get the problem but haven't dived too deep into different parsings....Would Maven's default fix the parsing issue?


Example:
```
<plugin>
                <groupId>pl.project13.maven</groupId>
                <artifactId>git-commit-id-plugin</artifactId>
                <version>2.2.1</version>
                <executions>
                    <execution>
                        <id>get-the-git-infos</id>
                        <goals>
                            <goal>revision</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>validate-the-git-infos</id>
                        <goals>
                            <goal>validateRevision</goal>
                        </goals>
                        <phase>package</phase>
                    </execution>
                </executions>
                <configuration>
                    <dateFormat>dd.MM.yyyy '@' HH:mm:ss z</dateFormat>
                    .....
                </configuration>
```

**Comment by TheSnoozer** [comment]
Default format is now `yyyy-MM-dd'T'HH:mm:ssZ`
Will be released in 2.2.3

behaviour still can be changed by using
``<dateFormat>dd.MM.yyyy '@' HH:mm:ss z</dateFormat>``
inside the plugin's configuration
